modules/snooker/src/main/ts/ephox/snooker/api/Generators.ts

```diff
@@ -166,12 +171,52 @@ const merging = function (generators: Generators) {
       // Remove any width calculations because they are no longer relevant.
       Css.remove(raw, 'width');
       Css.remove(cell, 'width');
+
+      const scope = Attribute.get(cell, 'scope');
+
+      if (Type.isString(scope)) {
+        Attribute.set(raw, 'scope', scope);
+      }
+
       return raw;
     };
   };
 
+  const merge = (cells: SugarElement[]) => {
+    const getScopeProperty = () => {
+      const baseScope = Attribute.get(cells[0], 'scope');
+      const scopes = [ 'row', 'col' ];
+
+      const isMixed = Arr.exists(cells.slice(1), (c) => {
+        const scopeOpt = Attribute.getOpt(c, 'scope');
+
+        return scopeOpt.fold(
+          Fun.never,
+          (scope) => Arr.contains(scopes, scope) && scope !== baseScope
+        );
+      });
+
+      return isMixed ? 'col' : baseScope;
+    };
+
+    Css.remove(cells[0], 'width');
+
+    const scope = getScopeProperty();
+
+    if (scope === 'col') {
+      Attribute.set(cells[0], 'scope', 'colgroup');
+    } else if (scope === 'row') {
+      Attribute.set(cells[0], 'scope', 'rowgroup');
```

Comment:
Is this too much simplification?

```suggestion
      Attribute.set(cells[0], 'scope', `${getScopeProperty()}group`);
```

Reply:
This change would replace all of the if statements etc, both here and above

Reply:
This wouldn't have worked, as `getScopeProperty` could here return `col|row` and `undefined`.
Some reworks of the code has given similar code though.